### ISSUE TITLE
gateway_shutdown() raises unecessary None exception

### ISSUE THREAD

**Issue by bartdag** [bartdag/py4j#175]
Reported by agronholm:

```
The following code:
 

import logging

from py4j.java_gateway import launch_gateway, JavaGateway, GatewayParameters


logging.basicConfig(level=logging.DEBUG)
port = launch_gateway()
params = GatewayParameters(port=port, eager_load=True)
gw = JavaGateway(gateway_parameters=params)
gw.shutdown()

Produces the following:

DEBUG:py4j.java_gateway:Exception while shutting down a socket

Traceback (most recent call last):

  File "/home/alex/virtualenv/asphalt/lib/python3.4/site-packages/py4j/java_gateway.py", line 301, in quiet_shutdown

    socket_instance.shutdown(socket.SHUT_RDWR)

OSError: [Errno 9] Bad file descriptor

INFO:py4j.java_gateway:Exception while shutting down callback server

Traceback (most recent call last):

  File "/home/alex/virtualenv/asphalt/lib/python3.4/site-packages/py4j/java_gateway.py", line 1428, in shutdown_callback_server

    self._callback_server.shutdown()

AttributeError: 'NoneType' object has no attribute 'shutdown'

DEBUG:py4j.java_gateway:Exception while closing

Traceback (most recent call last):

  File "/home/alex/virtualenv/asphalt/lib/python3.4/site-packages/py4j/java_gateway.py", line 290, in quiet_close

    closable.close()

AttributeError: 'NoneType' object has no attribute 'close'
```

**Comment by Copper-Head** [comment]
I'm also seeing this behavior. Any eta on a clean fix?

**Comment by bartdag** [comment]
Does the code in @master still produce this behavior? It should be fixed with 591f53e

**Comment by Copper-Head** [comment]
Feel free to close this, I can't reproduce :(
